### ISSUE TITLE
Upgrading from 5.2.11 -> 6.1.18 breaks projection

### ISSUE THREAD

**Issue by falconmick** [visgl/react-map-gl#1681]
#### Description
I have a basic Map with a SVGOverlay, when I was on 5.2.11 the svg I render perfectly lines up with the coordinates that I've placed for my polygon (using the project func that is passed to me via redraw.

Now that I have updated each time I zoom out the projection drifts to the right at a consistent pace and it's very obvious that the polygon no longer marries up to where it's supposed to be.

#### Environment (please complete the following information):

- Library Version: 6.1.18
- Mapbox Version: 2.6.1
- React Version: 17.0.2
- Browser Version: Edge 94.0.992.50
- OS: [e.g. Mac OS X 11.2] windows 10

**Comment by falconmick** [comment]
Just tested out using the context to get viewport and now it works correctly. It appears that whatever projection code the overlay redraw provides is not the same projection code that the context.viewport.project supplies

**Comment by falconmick** [comment]
And now that one isn't work either :(

**Comment by falconmick** [comment]
Yikes maybe it's the codebase I am in that is at fault, if I forget to get back to this post close it ;(

**Comment by falconmick** [comment]
Ok think I figured it out! There was some ever so slight resizing of the area in which the map sits on first navigation from the home page to the tab which has the map in it. Because I defined width and height as 100%, this library set the width and height to be a fixed px value based on the on loan width and height.

Then after first onViewportChange callback the library overwrote my 100% values with the new hard coded values it found and after that my page slightly shrunk and caused the coordinates to no longer marry up.

The solution is to destructure the viewport first and then after the destructure place the width="100%" and height=100%" so that they overwrite the px based values that are now on the viewport object 

Hope this helps somebody
